packages/core/src/http-client/http-client.ts

```diff
@@ -24,18 +24,20 @@ import {
  * If a destination name (and a JWT) are provided, it will try to resolve the destination.
  *
  * @param destination - A destination or a destination name and a JWT.
+ * @param customHeaders - Custom default headers for the resulting HTTP request.
  * @returns A [[DestinationHttpRequestConfig]].
  */
 export async function buildHttpRequest(
-  destination: Destination | DestinationNameAndJwt
+  destination: Destination | DestinationNameAndJwt,
+  customHeaders?: MapType<any>
```

Comment:
just a general remark for version 2.0. If we have a build method taking multiple potentially optional arguments the good old builder pattern is always an option instead of having too many arguments. Here still ok wit 3 ones. Destination and DestiantionNameAndJwt are actually two arguments.